tests/NCronJob.Tests/RunDependentJobTests.cs

```diff
@@ -285,6 +285,21 @@ public async Task ConfiguringDifferentDependentJobsForSchedulesShouldResultInInd
         results.ShouldContain("2");
     }
 
+    [Fact]
+    public async Task WhenJobIsNotCreated_DependentFailureJobShouldRun()
```

Comment:
I wonder if this wouldn't require a slight comment in the [doc](https://docs.ncronjob.dev/features/model-dependencies/#modelling-dependencies-with-executewhen).

Maybe something along the lines of 

```
`faulted` hook will be triggered when the job hasn't been able to be resolved or
when a issue prevented it to complete, despite the automated retries.
```

Reply:
Yeah let's do that.